include/seqan3/alignment/matrix/alignment_optimum.hpp

```diff
@@ -14,74 +14,119 @@
 
 #include <type_traits>
 
-#include <seqan3/alignment/matrix/alignment_coordinate.hpp>
+#include <seqan3/alignment/matrix/detail/matrix_coordinate.hpp>
 #include <seqan3/core/concept/core_language.hpp>
 #include <seqan3/core/type_traits/template_inspection.hpp>
+#include <seqan3/core/simd/concept.hpp>
+#include <seqan3/core/simd/simd_algorithm.hpp>
+#include <seqan3/core/simd/simd_traits.hpp>
 #include <seqan3/std/concepts>
 
 namespace seqan3::detail
 {
 
-/*!\brief Stores the current optimum of the alignment algorithms.
+/*!\brief Stores the current optimum of the alignment algorithm.
  * \ingroup alignment_matrix
- * \tparam score_t The type of the tracked alignment score.
+ *
+ * \tparam score_t The type of the tracked alignment score; must model either seqan3::arithmetic or
+ *                 seqan3::simd_concept.
  *
  * \details
  *
- * This is an aggregate type, so the score needs to be passed before the seqan3::alignment_coordinate during
- * construction.
+ * Stores the optimal score of the alignment computation and the corresponding indices of the cell with the optimal
+ * score within the alignment matrix.
+ * In case the optimum is used for the vectorised alignment computation this optimum stores the optimal scores and the
+ * respective cells as simd vectors.
  */
-template <arithmetic score_t>
+template <typename score_t>
 struct alignment_optimum
+#if SEQAN3_DOXYGEN_ONLY(1)0
 {
-    //!\brief The optimal score.
+    //!\brief The index type used to store the alignment coordinates of the optimum.
+    using index_t = IMPLEMENTATION_DEFINED;
```

Comment:
Whatis this IMPLEMENTATION_DEFINED?

Reply:
it is an implementation defined type. For arithmetic score types it is size_t and for simd types it is the simd type. But you don't care really, you just know that it depends on the implementation.